src/Service/FirstRegistrationService.php

```diff
@@ -335,7 +335,12 @@ public function zgwToFirstRegistrationHandler(array $data, array $configuration)
         // Merge the values and documents array.
         $mappingArray = array_merge($valuesArray, $documents);
 
-        $dataImportArray['name'] = 'Eerste Inschrijving ZDS';
+        if ($zaakObject->getValue('zaaktype')->getValue('identificatie') === 'B334') {
+            $dataImportArray['name'] = 'Eerste Inschrijving Expat ZDS';
+        } else {
+            $dataImportArray['name'] = 'Eerste Inschrijving ZDS';
+        }
```

Comment:
1. What if zaaktype (or identificatie) is null?
2. you could remove (/move) the else statement

Reply:
`
$dataImportArray['name'] = 'Eerste Inschrijving ZDS';
if ($zaakObject->getValue('zaaktype')->getValue('identificatie') === 'B334') {
  $dataImportArray['name'] = 'Eerste Inschrijving Expat ZDS';
}
`